sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilder.java

```diff
@@ -927,6 +939,9 @@ public EncryptedBlobClientBuilder requiresEncryption(boolean requiresEncryption)
      */
```

Comment:
I think we need to document that this throws now for `IllegalArgumentException` if encryption version is not V2_1. It should probably be similar to what we have here: https://github.com/Azure/azure-sdk-for-java/blob/afe363ca303fc21c3ad82828323c06915566a088/sdk/storage/azure-storage-blob-cryptography/src/main/java/com/azure/storage/blob/specialized/cryptography/EncryptedBlobClientBuilder.java#L513